data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/handler/EndpointDataReferenceHandler.java

```diff
@@ -67,30 +73,27 @@ public boolean canHandle(@NotNull MultipartRequest multipartRequest) {
      * - finally apply {@link EndpointDataReferenceReceiver} to the resulting EDR to dispatch it into the consumer environment.
      */
     @Override
-    public @Nullable MultipartResponse handleRequest(@NotNull MultipartRequest multipartRequest, @NotNull ClaimToken claimToken) {
+    public @NotNull MultipartResponse handleRequest(@NotNull MultipartRequest multipartRequest) {
+        Objects.requireNonNull(multipartRequest);
+        
+        // Read and transform the endpoint data reference from the request payload
         var edr = typeManager.readValue(multipartRequest.getPayload(), EndpointDataReference.class);
         var transformationResult = transformerRegistry.transform(edr);
         if (transformationResult.failed()) {
             monitor.severe("EDR transformation failed: " + String.join(", ", transformationResult.getFailureMessages()));
-            return createErrorMultipartResponse(multipartRequest);
+            return createMultipartResponse(badParameters(multipartRequest.getHeader(), connectorId));
         }
 
         var transformedEdr = transformationResult.getContent();
 
+        // Apply all endpoint data reference receivers to the endpoint data reference
         var receiveResult = receiverRegistry.receiveAll(transformedEdr).join();
```

Comment:
not for this PR, but `join`s should be avoided, we should think about making the handlers async to avoid blocking code